src/Microsoft.Extensions.Configuration.UserSecrets/PathHelper.cs

```diff
@@ -67,5 +71,24 @@ public static string GetSecretsPathFromSecretsId(string userSecretsId)
                 return Path.Combine(root, ".microsoft", "usersecrets", userSecretsId, Secrets_File_Name);
             }
         }
+
+        private static string InternalGetSecretsPath(string filePath)
+        {
+            if (filePath == null)
+            {
+                throw new ArgumentNullException(nameof(filePath));
+            }
+
+            var obj = JObject.Parse(File.ReadAllText(filePath));
```

Comment:
The main idea behind using IFileProvider is to abstract reading the file instead of using File.ReadAll directly.  PathHelper should be going through the FileProvider everywhere...